cmd/fleetctl/apple_mdm.go

```diff
@@ -44,124 +52,136 @@ func appleMDMCommand() *cli.Command {
 	}
 }
 
-func appleMDMSetupCommand() *cli.Command {
+func generateCommand() *cli.Command {
 	return &cli.Command{
-		Name:  "setup",
-		Usage: "Setup commands for Apple MDM",
+		Name: "generate",
+		// TODO: Remove when Apple MDM is production ready.
+		Hidden: true,
+		Flags: []cli.Flag{
+			configFlag(),
+			contextFlag(),
+			debugFlag(),
+		},
 		Subcommands: []*cli.Command{
-			appleMDMSetupSCEPCommand(),
-			appleMDMSetupAPNSCommand(),
-			appleMDMSetupDEPCommand(),
+			generateMDMAppleCommand(),
 		},
 	}
 }
 
-func appleMDMSetupSCEPCommand() *cli.Command {
-	// TODO(lucas): Define workflow when SCEP CA certificate expires.
-	var (
-		validityYears      int
-		cn                 string
-		organization       string
-		organizationalUnit string
-		country            string
-	)
+func generateMDMAppleCommand() *cli.Command {
 	return &cli.Command{
-		Name:  "scep",
-		Usage: "Create SCEP certificate authority",
+		Name:  "mdm-apple",
+		Usage: "Generates certificate signing request (CSR) and key for Apple Push Notification Service (APNs) and certificate and key for Simple Certificate Enrollment Protocol (SCEP) to turn on MDM features.",
 		Flags: []cli.Flag{
-			&cli.IntFlag{
-				Name:        "validity-years",
-				Usage:       "Validity of the SCEP CA certificate in years",
-				Required:    true,
-				Destination: &validityYears,
+			&cli.StringFlag{
+				Name:     "email",
+				Usage:    "The email address to send the signed APNS csr to.",
+				Required: true,
 			},
 			&cli.StringFlag{
-				Name:        "cn",
-				Usage:       "Common name to set in the SCEP CA certificate",
-				Required:    true,
-				Destination: &cn,
+				Name:     "org",
+				Usage:    "The organization requesting the signed APNS csr.",
+				Required: true,
 			},
```

Comment:
I wonder if we should just use the email as the org? I see two advantages

1) One less flag for the user to configure (cc @noahtalerman).

2) When Fleet verifies the user owns the email (by sending the signed request to the email), it would also verify that the user should be able to get a CSR with that subject.

Reply:
Actually, I dug into this further and found that CSRs that only include an email are happily accepted by Apple. My recommendation would be that we include only the email since that's the only thing that we can validate. I don't see any advantage to collecting or including any other fields. Does that sound good to y'all?

Reply:
Passing the org was a requirement in https://github.com/fleetdm/fleet/issues/7456. 

> $ fleetctl generate mdm-apple --email=mo@acme.com --org-name="Acme, Inc."

https://github.com/fleetdm/fleet/issues/7676 specifies that the org be saved by fleetdm.com

> fleetdm.com saves email and org name, and other data for those who request CSR from us.

afaik, apple doesn't require anything in the csr subject. However, I think you still have to provide something for it to be valid. It seemed like we were using the subject to include the email and org, instead of provided them as separate fields in the json request.

Reply:
@zwass @michalnicp my understanding is that we're collecting the org to create a record of organizations that are using MDM features in Fleet. (not because it's required by Apple)

@zhumo is this right?

Reply:
It is an apple requirement.

Reply:
Understood that we have requirements around this. Fine to proceed as it is, we should be aware of the fact that the email is verified, but the org is just trusting what the user provides.

Reply:
Yes absolutely. I will also periodically look at the orgs